pinot-controller/src/test/java/org/apache/pinot/controller/api/PinotInstanceRestletResourceTest.java

```diff
@@ -54,9 +57,17 @@ public void testInstanceListingAndCreation()
       throws Exception {
     // Check that there is only one CONTROLLER instance in the cluster
     String listInstancesUrl = _controllerRequestURLBuilder.forInstanceList();
-    JsonNode instanceList = JsonUtils.stringToJsonNode(sendGetRequest(listInstancesUrl));
-    assertEquals(instanceList.get("instances").size(), 1);
-    assertTrue(instanceList.get("instances").get(0).asText().startsWith(Helix.PREFIX_OF_CONTROLLER_INSTANCE));
+
+    TestUtils.waitForCondition(aVoid -> {
+      try {
+        String getResponse = sendGetRequest(listInstancesUrl);
+        JsonNode jsonNode = JsonUtils.stringToJsonNode(getResponse);
+        return jsonNode.get("instances").size() == 1 && jsonNode.get("instances").get(0).asText()
+            .startsWith(Helix.PREFIX_OF_CONTROLLER_INSTANCE);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
```

Comment:
The thrown exception will directly fail the test instead of retry. 
The cache data accessor may return null, which will exactly be caught here. 
Returning false would be good.

Reply:
We have to throw exception if getRequest fails for any arbitrary reason as in that case, retry might not help. Made the changes to check for null result though